File: src/main/java/com/kishlaly/ta/analyze/tasks/blocks/groups/FirstTrustModel_Buy_1.java
package com.kishlaly.ta.analyze.tasks.blocks.groups;

import com.kishlaly.ta.analyze.tasks.blocks.TaskBlockJava;
import com.kishlaly.ta.analyze.tasks.blocks.commons.ScreenSoftValidationJava;
import com.kishlaly.ta.analyze.tasks.blocks.two.Long_ScreenTwo_FirstTrustModelMainLogicJava;
import com.kishlaly.ta.utils.ContextJava;

import java.util.ArrayList;
import java.util.List;

public class FirstTrustModel_Buy_1 implements BlocksGroupJava {

    public List<TaskBlockJava> blocks() {
        ContextJava.TRIM_DATA = false;

        return new ArrayList<TaskBlockJava>() {{
            add(new ScreenSoftValidationJava());
            add(new Long_ScreenTwo_FirstTrustModelMainLogicJava());
        }};
    }

    @Override
    public String comments() {
        return "The First Trust Model";
    }
}

File: src/main/kotlin/com/kishlaly/ta/analyze/tasks/groups/trustmodel/FirstTrustModel_Buy_1.kt
package com.kishlaly.ta.analyze.tasks.groups.trustmodel

import com.kishlaly.ta.analyze.tasks.blocks.TaskBlock
import com.kishlaly.ta.analyze.tasks.blocks.commons.ScreenSoftValidation
import com.kishlaly.ta.analyze.tasks.blocks.complex.Long_ScreenTwo_FirstTrustModelMainLogic
import com.kishlaly.ta.analyze.tasks.groups.AbstractBlocksGroup
import com.kishlaly.ta.config.Context

class FirstTrustModel_Buy_1 : AbstractBlocksGroup() {
    override fun blocks(): List<TaskBlock> {
        Context.TRIM_DATA = false

        return object : ArrayList<TaskBlock>() {
            init {
                add(ScreenSoftValidation())
                add(Long_ScreenTwo_FirstTrustModelMainLogic())
            }
        }

    }

    override fun comments() = super.comments() + "The First Trust Model"
}